pages/_app.js

```diff
@@ -30,13 +37,19 @@ export default function App({ Component, pageProps }) {
   function handleLoadMore() {
     setPage((prevPage) => prevPage + 1);
   }
+  function handleAddEvents(newEvent) {
+    setOwnEvents([...ownEvents, { ...newEvent, id: uid() }]);
+  }
 
   return (
     <>
       <GlobalStyle />
       <Component
+        onAddEvent={handleAddEvents}
         apiData={data}
         handleLoadMore={handleLoadMore}
+        ownEvents={ownEvents}
+        setOwnEvents={setOwnEvents}
```

Comment:
You should not pass down the setter-function of your state. Instead you should use a handler function, which does the state update in your app component. The handler function you can then pass down to where you need it. You actually have done that with your `handleAddEvents` function. So, you can probably delete the `setOwnEvents` prop without any problem, if you do not use it anywhere.
```suggestion
   
```